chainer/optimizer.py

```diff
@@ -141,30 +143,42 @@ def state(self):
         """State dictionary."""
         return self._state
 
-    def add_hook(self, hook, name=None):
+    def add_hook(self, hook, name=None, timing='auto'):
         """Adds a hook function.
 
-        The hook function is called before any updates.
+        The hook function is called before or after any updates (see the timing
+        attribute).
 
         Args:
             hook (callable): Hook function to be added. It takes two
                 arguments: the update rule object and the parameter variable.
             name (str): Name of the hook function. The name attribute of the
                 hook function is used by default.
+            timing (str): Specifies when the hook is called. If 'auto', the
+                timimg property of the hook will decide the timing.
```

Comment:
Please write that `auto` is treated as `pre` if `timing` property is not available.

Reply:
Ok, sure.